src/main/java/com/varukha/onlinebookstore/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,21 @@
+package com.varukha.onlinebookstore.security;
+
+import com.varukha.onlinebookstore.repository.user.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username)
+            throws UsernameNotFoundException {
+        return userRepository.findUserByEmail(username)
+                .orElseThrow(() -> new RuntimeException("Can't find user by email"));
```

Comment:
It is bad practice to throw RuntimeException, better to create a custom or use an existing one but more specific